src/chelsa/bioclim.jl

```diff
@@ -11,23 +12,57 @@ Download [`CHELSA`](@ref) [`BioClim`](@ref) data from [chelsa-climate.org](https
     or `Symbol`s form `$(layerkeys(BioClim))`. Without a `layer` argument, all layers
     will be downloaded, and a `NamedTuple` of paths returned.
 
+# Keyword arguments
+$CHELSA_KEYWORDS
+
 Returns the filepath/s of the downloaded or pre-existing files.
 """
-getraster(T::Type{CHELSA{BioClim}}, layer::Union{Tuple,Int,Symbol}) = _getraster(T, layer)
+getraster(
+    T::Type{CHELSA{BioClim}}, 
+    layer::Union{Tuple,Int,Symbol}; 
+    version::Int = 2, 
+    patch::Int = latest_patch(T, Val(version))) = _getraster(T, layer, Val(version), patch)
 
-_getraster(T::Type{CHELSA{BioClim}}, layers::Tuple) = _map_layers(T, layers)
-_getraster(T::Type{CHELSA{BioClim}}, layer::Symbol) = _getraster(T, bioclim_int(layer))
-function _getraster(T::Type{CHELSA{BioClim}}, layer::Integer)
+_getraster(T::Type{CHELSA{BioClim}}, layers::Tuple, version, patch) = _map_layers(T, layers, version, patch)
+_getraster(T::Type{CHELSA{BioClim}}, layer::Symbol, version, patch) = _getraster(T, bioclim_int(layer), version, patch)
+function _getraster(T::Type{CHELSA{BioClim}}, layer::Integer, version, patch)
     _check_layer(T, layer)
-    path = rasterpath(T, layer)
-    url = rasterurl(T, layer)
+    path = rasterpath(T, layer, version, patch)
+    url = rasterurl(T, layer, version, patch)
     return _maybe_download(url, path)
 end
 
-rastername(::Type{CHELSA{BioClim}}, layer::Integer) = "CHELSA_bio$(layer)_1981-2010_V.2.1.tif"
+rastername(::Type{CHELSA{BioClim}}, layer::Integer, version::Val{2}, patch) = "CHELSA_bio$(layer)_1981-2010_V.2.$patch.tif"
+rastername(::Type{CHELSA{BioClim}}, layer::Integer, version::Val{1}, patch) = "CHELSA_bio10_$(lpad(layer, 2, "0")).tif"
+
+rasterpath(::Type{CHELSA{BioClim}}) = joinpath(rasterpath(CHELSA), "BioClim")
+rasterpath(T::Type{CHELSA{BioClim}}, layer::Integer, version, patch) = joinpath(rasterpath(T), rastername(T, layer, version, patch))
+
+rasterurl(::Type{CHELSA{BioClim}}, v::Val{2}) = joinpath(rasterurl(CHELSA, v), "climatologies/1981-2010/bio/")
+rasterurl(::Type{CHELSA{BioClim}}, v::Val{1}) = joinpath(rasterurl(CHELSA, v), "climatologies/bio/")
+rasterurl(T::Type{CHELSA{BioClim}}, layer::Integer, version, patch) = joinpath(rasterurl(T, version), rastername(T, layer, version, patch))
 
 rasterpath(::Type{CHELSA{BioClim}}) = joinpath(rasterpath(CHELSA), "BioClim")
 rasterpath(T::Type{CHELSA{BioClim}}, layer::Integer) = joinpath(rasterpath(T), rastername(T, layer))
 
-rasterurl(::Type{CHELSA{BioClim}}) = joinpath(rasterurl(CHELSA), "chelsa_V2/GLOBAL/climatologies/1981-2010/bio/")
-rasterurl(T::Type{CHELSA{BioClim}}, layer::Integer) = joinpath(rasterurl(T), rastername(T, layer))
+### Bioclim+
+layers(::Type{CHELSA{BioClimPlus}}) = layers(BioClimPlus)
+
+getraster(
+    T::Type{CHELSA{BioClimPlus}}, 
+    layer::Union{Tuple,Int,Symbol}; 
+    version::Int = 2, 
+    patch::Int = latest_patch(T, Val(version))) = _getraster(T, layer, Val(version), patch)
```

Comment:
Try not to lift runtime values to the type domain if we don't need to. Cant we just use a conditional on `version` or interpolate it into the url?

Reply:
The url is a bit different between the versions, but we can just use some if-else.

Reply:
We could even put the version in the CHELSA type then? And no version is the default? 

Introducing keywords means Rasters.jl has to forward them.

Reply:
Is it a problem to forward the keywords? If we just add `version` and `patch` to `getraster_keywords`, that should just work.

Reply:
I forgot we had added `getraster_keywords` for handling this. It was much more annoying before that.